assets/activation-scripts/activate.d/zdotdir/.zshrc

```diff
@@ -3,9 +3,25 @@
 #
 # See README.md for more information on the initialization process.
 
-# Save and restore the current tracelevel in the event that sourcing
-# bashrc launches an inner nested activation which unsets it.
+# Save environment variables that could be set if sourcing zshrc launches an
+# inner nested activation.
 _save_flox_activate_tracelevel="$_flox_activate_tracelevel"
+_save_FLOX_ENV="$FLOX_ENV"
+_save_FLOX_ORIG_ZDOTDIR="$FLOX_ORIG_ZDOTDIR"
+_save_ZDOTDIR="$ZDOTDIR"
+_save_FLOX_ZSH_INIT_SCRIPT="$FLOX_ZSH_INIT_SCRIPT"
+_save_add_env="$_add_env"
+_save_del_env="$_del_env"
+
+restore_saved_vars() {
+    export _flox_activate_tracelevel="$_save_flox_activate_tracelevel"
+    export FLOX_ENV="$_save_FLOX_ENV"
+    export FLOX_ORIG_ZDOTDIR="$_save_FLOX_ORIG_ZDOTDIR"
+    export ZDOTDIR="$_save_ZDOTDIR"
+    export FLOX_ZSH_INIT_SCRIPT="$_save_FLOX_ZSH_INIT_SCRIPT"
+    export _add_env="$_save_add_env"
+    export _del_env="$_save_del_env"
```

Comment:
I wonder if it would be better to pass all args in a single file referenced by an env var rather than having to do save and restore for everything

Reply:
Or regenerate ZDOTDIR every activation or something

Reply:
Probably, but I reckon we've exhausted our 2pts of work on this ticket. Can we tack it onto the end of the activation work as greenkeeping?

Reply:
💯